### ISSUE TITLE
Dynamic SSL sub-certificate expirations too long

### ISSUE THREAD

**Issue by marc-whil** [zaproxy/zaproxy#4673]
When I use dynamic ssl certificates for mitm, I'm getting expirations that go out over 8 years, which is causing Android Chrome to complain with NET::ERR_CERT_VALIDITY_TOO_LONG.

I don't see an option to configure the expiration date in the zap options.

The zap root certificate has a proper expiration of 1 year, but the sub-certificates per site have the inappropriately long expiration.  The zap documentation says "Every dynamically generated certificate is valid for 1000 days.", but that doesn't seem to be true.

Can we get an option to configure the duration of sub-certificates?

**Comment by thc202** [comment]
Thanks for raising the issue!

**Comment by dessant** [comment]
This issue affects all recent clients, the maximum validity of certificates issued after March 1, 2018 is limited to 825 days, while zap issues intermediate certificates with a validity of [1000 days](https://github.com/zaproxy/zaproxy/blob/65afe6af651cc789e96a3079edd86dd025ca8da1/zap/src/main/java/org/parosproxy/paros/security/SslCertificateServiceImpl.java#L164).

https://www.trustzone.com/ssl-certificate-validity-is-now-capped-at-a-maximum-of-2-years/

**Comment by kingthorin** [comment]
Have you encountered a UA that rejects Zap's cert due to validity period?

**Comment by dessant** [comment]
Yes, i have tested with the latest version of Firefox on Arch, it errors out with ERR_CERT_VALIDITY_TOO_LONG. I'm also having issues with an Andorid emulator, running Android Pie.

**Comment by kingthorin** [comment]
Does Firefox give you the option to proceed anyway?

I agree it should be fixed, just trying to judge impact.

**Comment by dessant** [comment]
Yes, the error can be bypassed on Firefox. It causes a bigger problem when the entire system is proxied, I couldn't find a workaround for proxying the entire traffic of the OS on Android.

**Comment by dessant** [comment]
Burp has the same issue, but they use the validity of the root certificate for intermediates, so adding a custom certificate is their current workaround.

https://support.portswigger.net/customer/portal/questions/17538500-burp-suite-1-7-37-create-burp-certificate-less-than-39-month

**Comment by thc202** [comment]
@dessant which Firefox version is latest? (for the record)

**Comment by dessant** [comment]
I was using Firefox 69.0.1.

**Comment by thc202** [comment]
Thanks.

**Comment by kingthorin** [comment]
@dessant, a fix will be in the weekly that's released next Monday.

**Comment by thc202** [comment]
This will be provided with an add-on (included by default).
